waspc/src/Wasp/SemanticVersion.hs

```diff
@@ -64,6 +76,22 @@ data ComparatorSet = ComparatorSet (NonEmpty Comparator)
 instance Show ComparatorSet where
   show (ComparatorSet comps) = unwords $ show <$> NE.toList comps
 
+-- | We define concatenation of two comparator sets union union of their comparators.
+instance Semigroup ComparatorSet where
+  (ComparatorSet compsl) <> (ComparatorSet compsr) = ComparatorSet $ compsl <> compsr
```

Comment:
Shouldn't we be enforcing a union?
```haskell
v1 = Version 1 0 0
v2 = Version 2 0 0
r1 = Range [gt v1, lt v2]
r2 = Range [lt v2]

-- results in Range [gt v1, lt v2, lt v2], I'd expect Range [gt v1, lt v2]
unionRange = r1 <> r2 
```

Reply:
Added basic nubing at level of range and comparator sets!
It would be cool to have more advanced stuff, like proper reducing of the range, and we could certainly do it, but that would be a bit more work so better to leave that for later.